src/main/kotlin/com/learnspigot/bot/manager/ForumManager.kt

```diff
@@ -33,10 +34,23 @@ class ForumManager(private val bot: JDA, private val datastore: Datastore, priva
     init {
         logger.info("Initiating ForumManager")
         bot.listener<MessageReceivedEvent> {
-            if(it.author.isBot) return@listener
             val channel = it.channel
-            if(channel !is ThreadChannel) return@listener
-            if(channel.parentChannel.id != System.getenv("HELP_CHANNEL_ID")) return@listener
+            if (channel !is ThreadChannel) return@listener
+            if (channel.parentChannel.id != System.getenv("HELP_CHANNEL_ID")) return@listener
+            val threadChannel: ThreadChannel = channel
```

Comment:
Instead of having a second variable, you can do something like this with the first variable:
```kt
            val channel: ThreadChannel = it.channel as? ThreadChannel ?: return@listener
```
This tries to cast it to a thread channel and returns null if it fails. We then use the elvis operator to return the function if the casted object is null. This acheives the same functionality without bloated code.

Reply:
Oh cool!